### PR TITLE
feat(react-native): improve manual sourcemaps docs

### PR DESCRIPTION
**Pull request by maxkomarychev**
- update examples to be more specific
- add naming rules for bundle files
- add troubleshooting section

### CODE REVIEW

File: src/platforms/react-native/manual-setup/sourcemaps.mdx [hunk 1]
```diff
@@ -10,6 +10,10 @@ To get unminified stack traces for JavaScript code, source maps must be generate
 
 If you want to manually invoke the [react-native packager](https://github.com/facebook/metro), you can get source maps by passing `–sourcemap-output`. Then upload the generated source maps with `sentry-cli` with the command `upload-sourcemaps`.
 
+## CLI interface
```
Comment (PeloWriter) [inline src/platforms/react-native/manual-setup/sourcemaps.mdx:13]:
```suggestion
## Configure CLI
```

File: src/platforms/react-native/manual-setup/sourcemaps.mdx [hunk 2]
```diff
@@ -10,6 +10,10 @@ To get unminified stack traces for JavaScript code, source maps must be generate
...
+
+Visit [configuration guide](https://docs.sentry.io/product/cli/configuration/) to configure defaults for CLI commands.
```
Comment (PeloWriter) [inline src/platforms/react-native/manual-setup/sourcemaps.mdx:15]:
```suggestion
Configure the CLI by reviewing the [configuration guide](https://docs.sentry.io/product/cli/configuration/) for CLI commands.
```

File: src/platforms/react-native/manual-setup/sourcemaps.mdx [hunk 3]
```diff
@@ -62,3 +67,18 @@ Sentry.init({
   dist: DISTRIBUTION_NAME,
 });
 ```
+
+### Naming rules
```
Comment (PeloWriter) [inline src/platforms/react-native/manual-setup/sourcemaps.mdx:71]:
```suggestion
### Naming Rules
```

File: src/platforms/react-native/manual-setup/sourcemaps.mdx [hunk 4]
```diff
@@ -62,3 +67,18 @@ Sentry.init({
...
+
+Name of the file used for `--bundle-output` and `--bundle` must be set
+differently per platform:
```
Comment (PeloWriter) [inline src/platforms/react-native/manual-setup/sourcemaps.mdx:74]:
```suggestion
The name of the file used for `--bundle-output` and `--bundle` must be set per platform:
```

File: src/platforms/react-native/manual-setup/sourcemaps.mdx [hunk 5]
```diff
@@ -62,3 +67,18 @@ Sentry.init({
...
+
+- for ios it is `main.jsbundle`
+- for android it is `index.android.bundle`
```
Comment (PeloWriter) [inline src/platforms/react-native/manual-setup/sourcemaps.mdx:77]:
```suggestion
- for iOS, the file name must be set to `main.jsbundle`
- for Android, the file name must be set to `index.android.bundle`
```

File: src/platforms/react-native/manual-setup/sourcemaps.mdx [hunk 6]
```diff
@@ -62,3 +67,18 @@ Sentry.init({
...
+
+If sourcemaps are still not recognized - check warnings for crash event, e.g.:
```
Comment (PeloWriter) [inline src/platforms/react-native/manual-setup/sourcemaps.mdx:79]:
```suggestion
## Troubleshooting

If source maps are still not recognized, check for warnings similar to:
```

File: src/platforms/react-native/manual-setup/sourcemaps.mdx [hunk 7]
```diff
@@ -62,3 +67,18 @@ Sentry.init({
...
+
+![Missing sources warning](001-manual-sourcemaps.png)
+
+Combination of filename of `--bundle` and `--url-prefix` should be equal to
+the URL displayed in a warning.
```
Comment (PeloWriter) [inline src/platforms/react-native/manual-setup/sourcemaps.mdx:84]:
```suggestion
The combination of the filename for `--bundle` and `--url-prefix` should be equal to
the URL displayed in a warning.
```

**Review by PeloWriter — COMMENTED** [review]
Thank you for these updates - I've edited for clarity. I will leave it @jennmueng to review for technical accuracy.

File: src/platforms/react-native/manual-setup/sourcemaps.mdx [hunk 8]
```diff
@@ -32,8 +36,9 @@ node_modules/@sentry/cli/bin/sentry-cli releases \
     files RELEASE_NAME \
     upload-sourcemaps \
     --dist DISTRIBUTION_NAME \
-    --strip-prefix /path/to/project/root \
-    /path/to/your/files
+    --url-prefix app:/// \
```
Comment (jennmueng) [inline src/platforms/react-native/manual-setup/sourcemaps.mdx:39]:
```suggestion
   --strip-prefix /path/to/project/root
```

`url-prefix` is actually not needed but we need to keep `strip-prefix` to strip the prefix from the files.

File: src/platforms/react-native/manual-setup/sourcemaps.mdx [hunk 9]
```diff
@@ -32,8 +36,9 @@ node_modules/@sentry/cli/bin/sentry-cli releases \
...
+    --bundle index.android.bundle \
+    --bundle-sourcemap index.android.bundle.map
```
Comment (jennmueng) [inline src/platforms/react-native/manual-setup/sourcemaps.mdx:41]:
```suggestion
  --rewrite index.android.bundle index.android.bundle.map
```
This is incorrect, as `bundle-sourcemap` expects a RAM bundle, and using it for normal source map bundles will show a warning: 
Screen Shot 2021-03-25 at 1 49 23 AM
The original method of passing each file path was correct. We also should add the `--rewrite` flag for React Native sourcemaps.

File: src/platforms/react-native/manual-setup/sourcemaps.mdx (hunk 7 above)
Comment (jennmueng) [inline src/platforms/react-native/manual-setup/sourcemaps.mdx:84]:
```suggestion
The bundle filename needs to match the filename on the event (and shown on the warning) to correctly apply the source maps.
```

File: src/platforms/react-native/manual-setup/sourcemaps.mdx (hunk 4 above)
Comment (jennmueng) [inline src/platforms/react-native/manual-setup/sourcemaps.mdx:74]:
```suggestion
Name of the bundle files must be set differently per platform:
```

**Review by jennmueng — CHANGES_REQUESTED** [review]
Using the CLI command proposed would not remove the path prefix, and also show a warning every time the command is run, we should use the original method with `--rewrite` added on.

File: src/platforms/react-native/manual-setup/sourcemaps.mdx (hunk 9 above)
Comment (maxkomarychev) [reply src/platforms/react-native/manual-setup/sourcemaps.mdx:41]: @jennmueng I haven't seen this warning in my experiments neither I'm using RAM bundle. Which version of CLI has this warning?

Comment (maxkomarychev) [reply src/platforms/react-native/manual-setup/sourcemaps.mdx:41]: I'm using  `@sentry/cli@1.63.1` and this is my command for bundle:

```
	NODE_OPTIONS=--max_old_space_size=4096 yarn react-native bundle \
		--entry-file $(BUNDLE_ENTRY_FILE) \
		--bundle-output $(BUNDLE_OUTPUT_FILE) \
		--sourcemap-output $(BUNDLE_SOURCEMAPS_OUTPUT_FILE) \
		--assets-dest $(BUNDLE_OUTPUT_FOLDER) \
		--platform $(K_PLATFORM) \
		--dev $(K_BUNDLE_DEV) \
		--reset-cache
```

Comment (maxkomarychev) [reply src/platforms/react-native/manual-setup/sourcemaps.mdx:41]: anyways. this seems to be working too, my current command is:

```
node_modules/@sentry/cli/bin/sentry-cli \
    releases \
    --org <org> \
    --project <project> \
    files <my release> \
    upload-sourcemaps \
    --dist <my dist> \
    --strip-prefix $(pwd) \
    --rewrite app.bundle/main.jsbundle \
     app.bundle/code.bundle.map
```


I guess main confusion was how to specify bundle and sourcemaps 🤔 and in which order

thanks!

**Review by jennmueng — APPROVED** [review]
Great! Thank you for this @maxkomarychev!

**Comment by PeloWriter** [comment]
Thank you for these updates, @maxkomarychev.

**Pull request merged**